### DISCUSSION TITLE
FIPS

### DISCUSSION THREAD

**Discussion by yoyomongo** [requarks/wiki#7233] [Help / Questions] [answered]
Is it possible to run on a fips compliant machine? During initialization I get a error0308010cdigital-envelope-routinesunsupported. When I temporarily disable fips, I am able to finish the setup. However, upon re-enabling fips, users are not able to log in. This is running on a rhel 8 box.

**Comment by NGPixel** [comment]
I have no idea what this means...

**Reply by yoyomongo** [reply]
> I have no idea what this means...

2 second google search: "The Federal Information Processing Standards (FIPS) of the United States are a set of publicly announced standards that the National Institute of Standards and Technology (NIST)". The error I'm getting above essentially means it's not utilizing the latest crypto standards. (Mostly likely md5 being used somewhere is my guess from googling both digital-envelope errors). Further troubleshooting I moved the database from postgres to edb running in postgres mode, which spits out a similar error to my original post error:060800A3:digital envelope routines:EVP_DigestInit_ex:disabled for fips after trying to finalize installation. This error spits out after "generating certificates"

**Reply by NGPixel** [reply]
Certificates are generated here:
https://github.com/requarks/wiki/blob/main/server/setup.js#L163
which is done using aes-256-cbc.

Node.js [can be configured to use FIPS](https://nodejs.org/api/crypto.html#fips-mode) but you're on your own going down this path.

**Reply by yoyomongo** [reply, marked as answer]
> Certificates are generated here: https://github.com/requarks/wiki/blob/main/server/setup.js#L163 which is done using aes-256-cbc.
> 
> Node.js [can be configured to use FIPS](https://nodejs.org/api/crypto.html#fips-mode) but you're on your own going down this path.

Thanks for the help! I was able to resolve the issue by changing privatekey type from pkcs1 to pkcs8 in the setup script
